packages/calling/src/CallingClient/calling/call.test.ts

```diff
@@ -361,6 +401,105 @@ describe('Call Tests', () => {
     call['callStateMachine'].state.value = 'S_SEND_CALL_PROGRESS';
     call.answer(localAudioStream);
     expect(call['callStateMachine'].state.value).toBe('S_SEND_CALL_CONNECT');
+
+    expect(mediaMetricSpy.mock.calls).toEqual([
+      [
+        METRIC_EVENT.MEDIA,
+        'S_ROAP_IDLE',
+        METRIC_TYPE.BEHAVIORAL,
+        call.getCallId(),
+        call.getCorrelationId(),
+        undefined,
+        undefined,
+        undefined,
+      ],
+      [
+        METRIC_EVENT.MEDIA,
+        MEDIA_EFFECT_ACTION.BNR_ENABLED,
+        METRIC_TYPE.BEHAVIORAL,
+        call.getCallId(),
+        call.getCorrelationId(),
+        undefined,
+        undefined,
+        undefined,
+      ],
+    ]);
+  });
+
+  it('testing enabling/disabling the BNR on an active call', async () => {
+    const localAudioStream = new MockMediaStream() as unknown as MediaSDK.LocalMicrophoneStream;
```

Comment:
If we are using onSpy, can we just do this instead of having a class inheriting from Emitter? The point of inheriting Emitter was to do .emit() and catch the event on .on()
Similar to mockStream
```
const localAudioStream = jest.fn().mockImplementation(() => {
    on: () => {};
    getEffect: () => mockEffect
});
```

Reply:
Reverted back